transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -699,14 +698,15 @@ def _parse_v2_advertisement(self, rssi, sender, data):
                     rotation_interval_power=EPHEMERAL_KEY_CYCLE_POWER,
                     current_timestamp=timestamp)
             except NotFoundError:
-                self._logger.warning("Key type {} is not found".format(key_type))
+                self._logger.warning("Key type {} is not found".format(key_type), exc_info=True)
                 return info, timestamp, None, None, None, None, None
 
             nonce = generate_nonce(device_id, timestamp, reboot_low, reboot_high_packed, counter_packed)
 
             try:
-                decrypted_data = decrypt_payload(key, data[7:], nonce)
-            except ValueError:
+                from Crypto.Cipher import AES
+                decrypted_data = decrypt_payload(AES, key, data[7:], nonce)
+            except (ImportError, ValueError):
                 self._logger.warning("Advertisement packet is not verified", exc_info=True)
```

Comment:
If left as is, this would log a warning for every encrypted broadcast packet received, which could be hundreds per second.

I would recommend refactoring this as:

```python
# At module level
try:
    from Crypto.Cipher import AES
    _HAS_CRYPTO = True
except import:
   _HAS_CRYTO = False


# inside DeviceAdapter __init__()
def __init__():
    if not _HAS_CRYPTO:
        self._logger.warning("pycryptodome is not installed, encrypted v2 broadcasts will be dropped.")


# inside _parse_v2_advertisement
def __parse_v2_advertisement():

    if is_encrypted and not _HAS_CRYTPO:
       return # Whatever return value indicates the broadcast was invalid
```

Reply:
You are right, thank you for suggestion, I wasn't sure how to properly log error case